### PR TITLE
Refactor/#73 modify players herd

### LINKED ISSUE #73: Refactor/#73 - modify players herd

### PR DESCRIPTION
**Pull request by vieraboschkova**
closes #73

### CODE REVIEW

**Comment by vieraboschkova** [comment]
closes #73 
unused imports to be removed after full refactor

File: src/Interfaces/GameConfigInterface.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  bankInitialStock: number;
+  dice?: { diceNumber: number; probability: number }[];
+}
+
+export interface predatorAnimalInterface {
+  name: AnimalNames;
+  kills: AnimalNames[];
+  isChasedAwayBy: [AnimalNames];
+  dice?: { diceNumber: number; probability: number }[];
+}
```
Comment (Enessetere) [inline src/Interfaces/GameConfigInterface.ts:14]: Consider adding protectorAnimalInterface, also split these interfaces into separate files.

File: src/Player.ts [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+  constructor(
+    name: string,
+    avatar = 'path to default avatar',
+    playersHerdConfig: {
+      name: AnimalNames;
+      tradeValue: Value;
+      role: AnimalRoles;
+      path: string;
+      initialStock: number;
+    }[],
```
Comment (Enessetere) [inline src/Player.ts]: Consider using here interface type, not type declaration. Adjust interface if necessary.

File: src/app/View.ts [hunk 3]
```diff
@@ -79,10 +79,10 @@ export class View {
...
     });
     const gameController = new GameController(this);
-    gameController.initializePlayer(
-      playersChosenName,
-      playersChosenAvatarPath,
-    );
+    // gameController.initializePlayer(
+    //   playersChosenName,
+    //   playersChosenAvatarPath,
+    // );
```
Comment (Enessetere) [inline src/app/View.ts]: Consider removing unnecessary comments, as initializePlayer is no longer available in GameController.

File: src/app/logic/Bank.ts [hunk 4]
```diff
@@ -1,9 +1,21 @@
-import { Player } from '~src/Player';
-import { Herd } from './Herd';
+import { Value } from '../../Animals/Animal';
+import { AnimalNames } from '../../Enums/AnimalNamesEnum';
+import { AnimalRoles } from '../../Enums/AnimalRolesEnum';
+import { Player } from '../../Player';
+// import { Herd } from './Herd';
```
Comment (Enessetere) [inline src/app/logic/Bank.ts]: Consider removing unnecessary comment

File: src/app/logic/Bank.ts [hunk 5]
```diff
@@ -1,9 +1,21 @@
...
-    super('Bank');
-    this.herd = new Herd(60, 24, 20, 12, 4, 4, 2);
+  constructor(
+    banksHerdConfig: {
+      name: AnimalNames;
+      tradeValue: Value;
+      role: AnimalRoles;
+      path: string;
+      initialStock: number;
+    }[],
```
Comment (Enessetere) [inline src/app/logic/Bank.ts]: Consider using interace in contructor. Adjust if necessary.

File: src/app/logic/Herd.ts [hunk 6]
```diff
@@ -1,65 +1,151 @@
...
-    numberOfHorses = 0,
-    numberOfSmallDogs = 0,
-    numberOfBigDogs = 0,
+    playersHerdConfig: {
+      name: AnimalNames;
+      tradeValue: Value;
+      role: AnimalRoles;
+      path: string;
+      initialStock: number;
+    }[],
```
Comment (Enessetere) [inline src/app/logic/Herd.ts]: Consider using herd config interface.

File: src/app/logic/Herd.ts [hunk 7]
```diff
@@ -1,65 +1,151 @@
...
-        animalData[1] += numberToAdd;
-      }
-      return animalData;
+  /**
+   * Finds the index of the tuple with given animal's name.
+   * @param {AnimalNames} animalName The animal name to search in animals tuples array.
+   * @return {number} The index of the tuple with searched animal.
+   */
+  private findAnimalTupleIndex(animalName: AnimalNames): number {
+    const indexOfAnimal = this.animals.findIndex((animal) => {
```
Comment (Enessetere) [inline src/app/logic/Herd.ts]: Destructuring will improve readability

File: test/herd.spec.ts [hunk 8]
```diff
@@ -1,35 +1,41 @@
...
+// import { Rabbit } from '../src/Animals/Rabbit';
+// import { Sheep } from '../src/Animals/Sheep';
+// import { SmallDog } from '../src/Animals/SmallDog';
 import { Wolf } from '../src/Animals/Wolf';
 import { AnimalNames } from '../src/Enums/AnimalNamesEnum';
 import { Herd } from '../src/app/logic/Herd';
+// import { AnimalRoles } from '../src/Enums/AnimalRolesEnum';
+import { mockHerdConfig } from '../src/app/logic/mockHerdConfig';
 
+// const newHerd = new Herd(mockHerdConfig);
```
Comment (Enessetere) [inline test/herd.spec.ts]: 1-16 - Remove unused imports and unnecessary comment

**Review by Enessetere — CHANGES_REQUESTED** [review]
Consider mentioned changes.
Why are you creating herdConfigInterface, if you are creating playerHerdInterface?

File: src/Animals/Animal.ts [hunk 9]
```diff
@@ -2,9 +2,10 @@ import { AnimalNames } from '../Enums/AnimalNamesEnum';
 import { AnimalRoles } from '../Enums/AnimalRolesEnum';
 import { pull } from 'lodash';
 
-type Value = 1 | 6 | 12 | 36 | 72;
+export type Value = 1 | 6 | 12 | 36 | 72;
 
-export abstract class Animal {
+// export abstract class Animal {
```
Comment (paula0403) [inline src/Animals/Animal.ts:7]: remove this comment

**Review by Enessetere — APPROVED** [review]
Looks good to me

File: src/Animals/Animal.ts (hunk 9 above)
Comment (vieraboschkova) [reply src/Animals/Animal.ts:7]: will do in refactor

**Review by paula0403 — APPROVED** [review]

**Review by paula0403 — APPROVED** [review]

**Pull request merged**